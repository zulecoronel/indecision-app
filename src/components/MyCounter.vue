<template>
  <section>
    <h2>Counter: {{ counter }}</h2>
    <h3 data-testid ="square-label">Square {{ squareCounter }}</h3>

    <div>
      <button class="btn" @click="decrementar">-1</button>
      <button class="btn" @click="incrementar">+1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter'

interface Props {
  value: number;
}

const props = defineProps<Props>();

const { counter, squareCounter } = useCounter(props.value)

// const squareCounter = computed(() => counter.value * counter.value)

const incrementar = () => {
  counter.value++
}

const decrementar = () => {
  counter.value--
}
</script>

<style lang="postcss" scoped>
.btn {
  @apply p-5 bg-blue-500 rounded hover:bg-blue-700 mr-2;
}
</style>
