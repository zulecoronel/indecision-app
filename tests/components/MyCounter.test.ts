import { describe, test } from 'vitest'
import { mount } from '@vue/test-utils'
import MyCounter from '../../src/components/MyCounter.vue'


describe('<MyCounter />', () => {
  test('should match snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 5
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  });

  test('renders the counter value correctly', () => {
    const value = 5;
    const wrapper = mount (MyCounter, {
        props: {
            value:value,
        },
    });
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);


   // console.log(wrapper.html())
  });

  test('increments the counter when +1 button is clicked', async () => {
    const value = 5;
    const wrapper = mount (MyCounter, {
        props: {
            value:value,
        },
    });

    const btnIncrement = wrapper.find('button');
    await btnIncrement.trigger('click');

    expect(counterLabel.text()).toContain(`Counter: ${value + 1 }`);
    expect(squareLabel.text()).toContain(`Square: ${(value + 1)*(value + 1) }`);

    });
});
