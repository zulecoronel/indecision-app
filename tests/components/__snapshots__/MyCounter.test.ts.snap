// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should match snapshot 1`] = `
"<section data-v-0352012b="">
  <h2 data-v-0352012b="">Counter: 5</h2>
  <h3 data-v-0352012b="">Square 25</h3>
  <div data-v-0352012b=""><button data-v-0352012b="" class="btn">-1</button><button data-v-0352012b="" class="btn">+1</button></div>
</section>"
`;
